libavformat/whip.c

```diff
@@ -1467,6 +1467,9 @@ static int on_rtp_write_packet(void *opaque, const uint8_t *buf, int buf_size)
         return 0;
     }
 
+    if (!(whip->udp->flags & AVIO_FLAG_NONBLOCK)) 
+        whip->udp->flags |= AVIO_FLAG_NONBLOCK;
+
```

Comment:
Does whip actually still directly write to the UDP socket after it has handed it over to the DTLS URLContext?
Does that not break the DTLS session?
In that case, yeah, it does need to set the nonblock flag to the flags it desires.

On that note, the if is entirely unnecessary. It's going to take more CPU cycles to check if the flag is set than it takes to simply set it. Though of course we're talking negligible amounts of cycles here.